indexprovider/wrapper_test.go

```diff
@@ -368,89 +345,109 @@ func TestUnsealedStateManagerStorageList(t *testing.T) {
-					SectorFileType: storiface.FTUnsealed,
-				},
-				{
-					SectorID:       sectorID,
-					SectorFileType: storiface.FTSealed,
-				},
-			}
-		},
-		expect: func(prov *mock_provider.MockInterfaceMockRecorder, prop market.DealProposal) {
-			// Expect only one call to NotifyPut with fast retrieval = true (unsealed)
-			// because we ignore a state change to cache
-			prov.NotifyPut(gomock.Any(), gomock.Any(), gomock.Any(), metadata.Default.New(&metadata.GraphsyncFilecoinV1{
-				PieceCID:      prop.PieceCID,
-				VerifiedDeal:  prop.VerifiedDeal,
-				FastRetrieval: true,
-			})).Times(1)
-		},
-	}, {
-		name: "unsealed and cached status",
-		storageListResponse: func(sectorID abi.SectorID) []storiface.Decl {
-			return []storiface.Decl{
-				{
-					SectorID:       sectorID,
-					SectorFileType: storiface.FTUnsealed,
-				},
-				{
-					SectorID:       sectorID,
-					SectorFileType: storiface.FTCache,
-				},
-			}
-		},
-		expect: func(prov *mock_provider.MockInterfaceMockRecorder, prop market.DealProposal) {
-			// Expect only one call to NotifyPut with fast retrieval = true (unsealed)
-			// because we ignore a state change to cache
-			prov.NotifyPut(gomock.Any(), gomock.Any(), gomock.Any(), metadata.Default.New(&metadata.GraphsyncFilecoinV1{
-				PieceCID:      prop.PieceCID,
-				VerifiedDeal:  prop.VerifiedDeal,
-				FastRetrieval: true,
-			})).Times(1)
-		},
-	}, {
-		name: "sealed and cached status",
-		storageListResponse: func(sectorID abi.SectorID) []storiface.Decl {
-			return []storiface.Decl{
-				{
-					SectorID:       sectorID,
-					SectorFileType: storiface.FTSealed,
-				},
-				{
-					SectorID:       sectorID,
-					SectorFileType: storiface.FTCache,
-				},
-			}
-		},
-		expect: func(prov *mock_provider.MockInterfaceMockRecorder, prop market.DealProposal) {
-			// Expect only one call to NotifyPut with fast retrieval = true (unsealed)
-			// because we ignore a state change to cache
-			prov.NotifyPut(gomock.Any(), gomock.Any(), gomock.Any(), metadata.Default.New(&metadata.GraphsyncFilecoinV1{
-				PieceCID:      prop.PieceCID,
-				VerifiedDeal:  prop.VerifiedDeal,
-				FastRetrieval: false,
-			})).Times(1)
-		},
-	}}
+		name         string
+		initialState func(sectorID abi.SectorID) []storiface.Decl
+		sus          func(sectorID abi.SectorID) map[abi.SectorID]db.SealState
+		expect       func(*mock_provider.MockInterfaceMockRecorder, market.DealProposal)
+	}{
+		{
+			name: "unsealed and sealed status",
+			sus: func(sectorID abi.SectorID) map[abi.SectorID]db.SealState {
+				return map[abi.SectorID]db.SealState{
+					sectorID: db.SealStateUnsealed,
+					sectorID: db.SealStateCache,
+				}
+			},
+			initialState: func(sectorID abi.SectorID) []storiface.Decl {
+				return []storiface.Decl{
+					{
+						SectorID:       sectorID,
+						SectorFileType: storiface.FTUnsealed,
+					},
+					{
+						SectorID:       sectorID,
+						SectorFileType: storiface.FTSealed,
+					},
+				}
+			},
```

Comment:
These fixtures need more work, they are terrible right now. A lot if indirection for setting up relatively basic tests.